blocks/library/freeform/index.js

```diff
@@ -13,22 +56,88 @@ registerBlock( 'core/freeform', {
 	category: 'common',
 
 	attributes: {
-		html: html(),
+		content: children(),
+		listType: internalValue( null ),
+		inBlockQuote: internalValue( false ),
+		editor: internalValue( null ),
+	},
+
+	defaultAttributes: {
+		content: <p />,
+		listType: null,
+		inBlockQuote: false,
+		editor: null,
 	},
 
-	edit( { attributes } ) {
+	controls: [
+		{
+			icon: 'editor-quote',
+			title: wp.i18n.__( 'Quote' ),
+			isActive: blockQuoteIsActive(),
+			onClick: execCommand( 'mceBlockQuote' ),
+		},
+		{
+			icon: 'editor-ul',
+			title: wp.i18n.__( 'Convert to unordered' ),
+			isActive: listIsActive( 'UL' ),
+			onClick: execCommand( 'InsertUnorderedList' ),
+		},
+		{
+			icon: 'editor-ol',
+			title: wp.i18n.__( 'Convert to ordered' ),
+			isActive: listIsActive( 'OL' ),
+			onClick: execCommand( 'InsertOrderedList' ),
+		},
+	],
+
+	advControls: [
+		{
+			icon: 'editor-outdent',
+			title: wp.i18n.__( 'Outdent list item' ),
+			onClick: execCommand( 'Outdent' ),
+		},
+		{
+			icon: 'editor-indent',
+			title: wp.i18n.__( 'Indent list item' ),
+			onClick: execCommand( 'Indent' ),
+		},
+	],
+
+	edit( { attributes, setAttributes, focus, setFocus } ) {
+		const { content } = attributes;
+
 		return (
-			<div
-				contentEditable
-				suppressContentEditableWarning
-			>
-				{ attributes.html }
-			</div>
+			<Editable
+				className="blocks-freeform"
+				value={ content }
+				getSettings={ ( settings ) => ( {
+					...settings,
+					plugins: ( settings.plugins || [] ).concat( 'lists' ),
+				} ) }
+				onSetup={ ( editor ) => {
+					editor.on( 'nodeChange', ( nodeInfo ) => {
+						setAttributes( {
+							listType: findListType( nodeInfo ),
+							inBlockQuote: findInBlockQuote( nodeInfo ),
```

Comment:
Like @nylen mentioned, we shouldn't use `setAttributes` for "temporary" state that is not going to be persisted in the block output markup. We should use local state for this (the `edit` here can be a regular React Component). 

You'll be able to access the local state in the controls if you embed the controls inside the `edit` function (see #830)